File: packages/postpin-python/src/postpin/resources/plans.py
from __future__ import annotations

from typing import List, Optional

from .._http import HTTPClient
from ..types import Plan


class Plans:
    def __init__(self, http: HTTPClient) -> None:
        self._http = http

    def list(self, *, timeout: Optional[float] = None) -> List[Plan]:
        """List all public, active subscription plans."""
        data, _ = self._http.request("GET", "/public/plans", timeout=timeout)
        return [Plan._from(item) for item in (data or [])]

File: packages/postpin-node/src/resources/plans.ts
import type { HttpClient } from "../http.js";
import type { Plan, RequestOptions } from "../types.js";

/** Public subscription plans (the ones shown on the pricing page). */
export class Plans {
  constructor(private readonly http: HttpClient) {}

  /** List all public, active plans in display order. */
  async list(options: RequestOptions = {}): Promise<Plan[]> {
    const { data } = await this.http.request<Plan[]>({
      method: "GET",
      path: "/public/plans",
      signal: options.signal,
    });
    return data;
  }
}
